extensions/cronet/src/main/java/com/google/android/exoplayer2/ext/cronet/CronetDataSource.java

```diff
@@ -682,6 +651,50 @@ protected UrlResponseInfo getCurrentUrlResponseInfo() {
 
   // Internal methods.
 
+  private int readIntoByteArray(byte[] buffer, int offset, int readLength) throws HttpDataSourceException {
```

Comment:
This new method is just the body of `read(byte[] buffer, int offset, int readLength)`, without the assertion, which remains in the public read method.

This new method `readIntoByteArray(byte[] buffer, int offset, int readLength)` is now called from `open()` to get the response body.